rust/crd/src/lib.rs

```diff
@@ -234,22 +234,40 @@ impl SparkApplication {
             .flat_map(|v| v.iter())
             .cloned()
             .collect();
+
+        self.check_mounts(&mut result, s3bucket);
+
+        result
+    }
+
+    fn check_mounts(&self, result: &mut Vec<VolumeMount>, s3bucket: &Option<InlinedS3BucketSpec>) {
```

Comment:
Naming is hard but I feel this function could be better called `add_common_volume_mounts` or similar. Also I would have it return the extended `&Vec<VolumeMount>` like this (not tested):
```suggestion
    fn add_common_volume_mounts(&self, mounts: &mut Vec<VolumeMount>, s3bucket: &Option<InlinedS3BucketSpec>) -> &Vec<VolumeMount> {
```

Reply:
Implemented as suggested.